scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/solver/PrioritySolver.java

```diff
@@ -513,6 +521,176 @@ private void satisfyGoalGeneral(Goal goal) throws SchedulingInterruptedException
     logger.info("Finishing goal satisfaction for goal " + goal.getName() +":"+ (missingConflicts.size() == 0 ? "SUCCESS" : "FAILURE. Number of conflicts that could not be addressed: " + missingConflicts.size()));
   }
 
+  private ConflictSolverResult solveMissingRecurrenceConflict(
+      final MissingRecurrenceConflict missingRecurrenceConflict,
+      final Goal goal
+      ) throws SchedulingInterruptedException
+  {
+    Optional<Long> maxIterations = Optional.empty();
+    Interval interval;
+    final var spaceToFill = (missingRecurrenceConflict.nextStart.minus(missingRecurrenceConflict.lastStart));
+    final var remainderMin = spaceToFill.remainderOf(missingRecurrenceConflict.minMaxConstraints.end);
+    final var remainderMax = spaceToFill.remainderOf(Duration.max(missingRecurrenceConflict.minMaxConstraints.start, MICROSECOND));
+    final var minNbActivities = spaceToFill.dividedBy(missingRecurrenceConflict.minMaxConstraints.end) - 1 + (remainderMin.isZero() ? 0 : 1);
+    final var maxNbActivities = spaceToFill.dividedBy(Duration.max(missingRecurrenceConflict.minMaxConstraints.start, MICROSECOND)) - 1+ (remainderMax.isZero() ? 0 : 1);
+    final var step = maxIterations.isPresent() ? (int) Math.floor((maxNbActivities - minNbActivities)/maxIterations.get()) : 1;
+    var nbActivities = minNbActivities;
+    Collection<SchedulingActivity> lastPartiallySat = null;
+    while(true){
+      //create netwrok with nbActivities
+      final var conflict = makeActivityNetworkConflict(missingRecurrenceConflict, nbActivities);
+      if(conflict.isPresent()) {
+        final var conflictResult = solveActivityNetworkConflict(conflict.get(), goal);
+        if (conflictResult.satisfaction().equals(ConflictSatisfaction.SAT)) {
+          return conflictResult;
+        } else {
+          plan.remove(conflictResult.activitiesCreated());
+          if(hasDuplicates(conflictResult.activitiesCreated().stream().toList())){
+            //duplicates means it is unsolvable and it is no use increasing the number of activities
+            break;
+          }
+          if(conflictResult.satisfaction().equals(ConflictSatisfaction.PARTIALLY_SAT)){
+            lastPartiallySat = conflictResult.activitiesCreated();
+          }
+        }
```

Comment:
If I'm interpreting this right, it looks like if solving the conflict with `n` activities fails, it starts over and tries again with `n+1`, instead of trying to continue with the existing activities. So it will potentially trigger another `n` duplicate simulations, with a worst case of `N^2` total simulations, where `N` is the final number of activities that actually succeeds.

Reply:
Or actually it would start over and try again with `n+step` activities, but `step` is based on `maxIterations`, which is never set, so `step` will always be `1`.

Reply:
You're right. Refactored so it uses previously inserted activities.